packages/transport/src/modules/messages/MessageController.ts

```diff
@@ -287,26 +287,48 @@ export class MessageController extends TransportController {
     @log()
     public async sendMessage(parameters: ISendMessageParameters): Promise<Message> {
         const parsedParams = SendMessageParameters.from(parameters);
-        if (!parsedParams.attachments) parsedParams.attachments = [];
 
+        if (!parsedParams.attachments) parsedParams.attachments = [];
+        /*
+        if (parameters.content.constructor.name === "Notification")
+        {
+*/
         const secret = await CoreCrypto.generateSecretKey();
         const serializedSecret = secret.serialize(false);
         const addressArray: ICoreAddress[] = [];
         const envelopeRecipients: MessageEnvelopeRecipient[] = [];
+        const peerDeletedAddressArray: string[] = [];
+        const missingOrInactiveRelationshipAddressArray: string[] = [];
         for (const recipient of parsedParams.recipients) {
-            const relationship = await this.relationships.getActiveRelationshipToIdentity(recipient);
-            if (!relationship) {
-                throw TransportCoreErrors.messages.missingOrInactiveRelationship(recipient.toString());
+            const relationship = await this.relationships.getRelationshipToIdentity(recipient);
+            if (
+                relationship &&
+                (relationship.status === RelationshipStatus.Terminated || relationship.status === RelationshipStatus.Active) &&
+                relationship.peerDeletionInfo?.deletionStatus !== "Deleted"
+            ) {
+                const cipherForRecipient = await this.secrets.encrypt(relationship.relationshipSecretId, serializedSecret);
+                envelopeRecipients.push(
+                    MessageEnvelopeRecipient.from({
+                        address: recipient,
+                        encryptedKey: cipherForRecipient
+                    })
+                );
+                addressArray.push(recipient);
+            }
+            if (!(relationship && (relationship.status === RelationshipStatus.Terminated || relationship.status === RelationshipStatus.Active))) {
+                missingOrInactiveRelationshipAddressArray.push(recipient.address);
+            }
+            if (relationship?.peerDeletionInfo?.deletionStatus === "Deleted") {
+                peerDeletedAddressArray.push(recipient.address);
             }
+        }
 
-            const cipherForRecipient = await this.secrets.encrypt(relationship.relationshipSecretId, serializedSecret);
-            envelopeRecipients.push(
-                MessageEnvelopeRecipient.from({
-                    address: recipient,
-                    encryptedKey: cipherForRecipient
-                })
-            );
-            addressArray.push(recipient);
+        if (peerDeletedAddressArray.length > 0) {
```

Comment:
Isn't it necessary to consider the case that the peer is "ToBeDeleted" in addition to the case that the peer is "Deleted" as well, as we do in the `SendMessage.ts`?

Reply:
No, because here I can't distinguish between Notification and Message etc. and so I only block what needs to be blocked for Notifications. The method sendMessage from MessageController is only called directly in cases where a notification is sent or  in case of Requests you call canCreate before with the necessary validations.